### PR TITLE
fix: toggle selectmenu via keyboard

### PR DESCRIPTION
**Pull request by gkatsev**
This makes preventClick apply to keypresses in media-chrome-button.

It also has selectmenu listen to keypress events on the button and the listbox so that it can open and close the menu as expected via Enter/Space and Escape.

### CODE REVIEW

File: src/js/experimental/media-chrome-selectmenu.js [hunk 1]
```diff
@@ -110,6 +110,36 @@ class MediaChromeSelectMenu extends window.HTMLElement {
...
+    if (!this.keysUsed.includes(key)) {
+      this.removeEventListener('keyup', this.#keyupListener);
+      return;
+    }
+
+    const isButton = e.composedPath().includes(this.#button);
+
+    // only allow Enter/Space on the button itself and not on the listbox
+    // and allow hiding the menu when pressing Escape when focused on the listbox
+    if (isButton && (key === 'Enter' || key === ' ')) {
```
Comment (luwes) [inline src/js/experimental/media-chrome-selectmenu.js:127]: if I use space to open the page scrolls down so we need a event.preventDefault() here

Comment (gkatsev) [reply src/js/experimental/media-chrome-selectmenu.js:127]: hm... not happening for me locally, but happy to add one. Has to go in the `keydownListener`, though.

Comment (gkatsev) [reply src/js/experimental/media-chrome-selectmenu.js:127]: updated.

**Review by luwes — CHANGES_REQUESTED** [review]
these still need fixing I think

- escape doesn't work if the focus is in the listbox
- the whole listbox can get focus somehow, see screenshot 
SCR-20230331-d6f
- I expect the listbox to close if I click out of the listbox, close on blur basically. but the video shouldn't start playing if the click is on the gesture receiver or the play button
- the bug of selecting 3 times Japanese for example is still there, it eventually selects English?

**Comment by gkatsev** [comment]
> * escape doesn't work if the focus is in the listbox

Hm... weird, I'll take another look but it should've been accounted for.

> * the whole listbox can get focus somehow, see screenshot

This is expected. When the listbox opens, it gets focus.

> I expect the listbox to close if I click out of the listbox, close on blur basically. but the video shouldn't start playing if the click is on the gesture receiver or the play button

This is a tricky one, since we can't prevent the action on other components from within the selectmenu itself. Though, we definitely can make it close on a click outside the component.
This probably deserves a larger media-chrome discussion, as we may get more things that we may want that handle this.

> the bug of selecting 3 times Japanese for example is still there, it eventually selects English?

This is being handled elsewhere. There's some work on this in #513.

**Comment by gkatsev** [comment]
> > * escape doesn't work if the focus is in the listbox
> 
> Hm... weird, I'll take another look but it should've been accounted for.

Weird, so, it works for the elements in the control bar but not the other ones. It's because that first CC button is still a chrome-selectmenu with a slotted captions-listbox. So, the fix is to make sure that the keypress listener is being re-added on a slotchange.

**Review by luwes — APPROVED** [review]

**Pull request merged**